frontend/src/lib/logic/promptLogic.tsx

```diff
@@ -55,23 +58,43 @@ export const prompt = kea({
                 failure && failure(err)
             }
         },
-    }),
-})
+    })),
+])
+
+type PromptProps = {
+    value: string
+    visible: ModalProps['visible']
+    afterClose: ModalProps['afterClose']
+    close: ModalProps['onCancel']
+    title: ModalProps['title']
+    modalProps: ModalProps
+    rules: FormItemProps['rules']
+    placeholder: InputProps['placeholder']
+}
 
 // Based on https://github.com/wangsijie/antd-prompt/blob/master/src/index.js
 // Adapted for ant.design v4 and added cancellation support
-function Prompt({ value, visible, afterClose, close, title, modalProps, rules, placeholder }) {
+function Prompt({
+    value,
+    visible,
+    afterClose,
+    close,
+    title,
+    modalProps,
+    rules,
+    placeholder,
+}: PromptProps): JSX.Element {
     const [form] = Form.useForm()
-    const onFinish = (values) => {
-        close(values.field)
+    const onFinish = (values: Record<string, any>): void => {
+        close && close(values.field)
```

Comment:
```suggestion
        close?.(values.field)
```

Maybe just personal preference but that reads easier IMHO